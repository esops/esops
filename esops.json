 [
   "github:sartaj/esops-ts#2019-10-18-8 --subdir ts-node",
   "github:sartaj/esops-ts#2019-10-18-8 --subdir ts-ramda",
   "./package.extend.json -o ./package.json --merge"
 ]