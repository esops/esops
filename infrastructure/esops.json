{
  "stack": ["github:sartaj/typescript-node-stack/stack#0.1.0", "./"],
  "destination": "../"
}
