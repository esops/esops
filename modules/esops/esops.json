<pipe-me />
<target-react-native-web module={'@my/app-navigation'} />
<semantic-design />