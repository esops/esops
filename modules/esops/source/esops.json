[
  "npm:sartaj/typescript-target-stack",
  "http:",
  "https:",
  "git:github.com:sartaj/typescript-stack.git",
  "github:sartaj/typescript-stack"
]
