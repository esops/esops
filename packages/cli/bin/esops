#!/usr/bin/env node

require('@esops/language-babel/node')

// run the CLI with the current process arguments
require('../')(process.argv)
