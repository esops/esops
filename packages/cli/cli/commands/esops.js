// eslint-disable-next-line

const path = require('path')
const target = require('@esops/target-react-native-web').default
// const publishGHPages = require('@esops/publish-github-pages')
const log = require('@esops/logger')

const messages = require('../messages')

const devStaticGithub = async context => {
  const { parameters, cwd } = context

  switch (parameters.second) {
    case 'web':
    case 'browser':
    case 'github-pages':
      await target({ devMode: true, cwd })
      break
    case 'help':
    default:
      log.md(messages.webAvailableCommands)
  }
}

const shipStaticGithub = async context => {
  const { parameters, cwd } = context

  const distribution = {
    devMode: false,
    buildPath: path.join(cwd, './.esops/target/web/dist')
  }

  switch (parameters.second) {
    case 'web':
    case 'browser':
    case 'github-pages':
      await target(distribution)
      await publishGHPages({ ...context, ...distribution })
      break
    case 'help':
    default:
      log.md('For `esops ship`, available commands are `github-pages`.')
  }
}

// eslint-disable-next-line
module.exports = {
  run: async context => {
    const { parameters } = context

    switch (parameters.first) {
      case 'dev':
        devStaticGithub(context)
        break
      case 'ship':
        shipStaticGithub(context)
        break
      case 'help':
      case undefined:
        await log.mdFile(path.join(__dirname, '../../docs/welcome.md'))
        break
      default:
        log.md('For `esops`, available commands are `dev, ship, & help`.')
    }
  }
}
