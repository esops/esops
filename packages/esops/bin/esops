#!/usr/bin/env node


require('@esops/compiler-babel/node')

// run the CLI with the current process arguments
require('../src/cli')(process.argv)
