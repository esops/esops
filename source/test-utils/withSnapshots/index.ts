import {curry} from 'ramda'
import * as path from 'path'
import * as fs from 'fs'
import * as riteway from 'riteway'

const updateSnapshot = (name, snapshotPath, actual) => {
  fs.writeFileSync(snapshotPath, actual)
  console.log(`snapshot ${name} has been updated`)
  return {
    expected: actual,
    actual
  }
}

const toMatchSnapshot = curry((snapshotDir, name, contents) => {
  const shouldUpdateSnapshots = process.env.UPDATE_SNAPSHOTS
  const snapshotPath = path.join(snapshotDir, `${name}.snap`)

  if (!fs.existsSync(snapshotDir)) fs.mkdirSync(snapshotDir)

  const actual =
    typeof contents !== 'string' ? JSON.stringify(contents, null, 2) : contents

  if (!fs.existsSync(snapshotPath)) {
    if (shouldUpdateSnapshots) return updateSnapshot(name, snapshotPath, actual)
    else
      throw new Error(
        'Snapshot does not exist yet. Run with UPDATE_SNAPSHOTS=1 to update snapshots.'
      )
  }

  const expected = fs.readFileSync(snapshotPath, {encoding: 'utf-8'})

  if (actual !== expected) {
    if (shouldUpdateSnapshots) {
      return updateSnapshot(name, snapshotPath, actual)
    } else {
      return {
        actual,
        expected
      }
    }
  } else {
    return {
      actual,
      expected
    }
  }
})

function hashCode(str) {
  return str
    .split('')
    .reduce(
      (prevHash, currVal) =>
        ((prevHash << 5) - prevHash + currVal.charCodeAt(0)) | 0,
      0
    )
}

export const withSnapshots = curry(
  (cwd: string | null, dirname: string | null, describeMessage, callback) => {
    const dirPath: string = cwd || process.cwd()
    const snapshotDirname: string = dirname || 'snapshots'
    const snapshotDir = path.join(dirPath, snapshotDirname)
    const matchByName = toMatchSnapshot(snapshotDir)

    const describeFunc = riteway.describe
    if (typeof describeFunc === 'function') {
      describeFunc(describeMessage, async assert => {
        const assertSnap = ({given, should, snap}) => {
          const name = `${describeMessage}-${hashCode(
            describeMessage + given + should
          )}`

          assert({
            given,
            should,
            ...matchByName(name, snap)
          })
        }
        callback(assert, assertSnap)
      })
    }
  }
)
