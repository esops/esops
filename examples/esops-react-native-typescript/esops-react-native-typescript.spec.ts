import * as path from 'path'

import esops from '../../index'

import {withSnapshots} from '../../test-utilities/withSnapshots'
import {withTempDir} from '../../test-utilities/withTempDir'
import {
  getFileContents,
  getSortedFilePaths
} from '../../test-utilities/fs-utils'

const describe = withSnapshots(__filename)

const rootPath = path.join(__dirname, './module')

describe('esops react native typescript', async assert => {
  await withTempDir(__dirname, rootPath, async root => {
    await esops({
      root: path.join(root, 'infrastructure'),
      logLevel: 'info',
      prompts: [true]
    })

    await assert({
      given: 'esops_react_native_typescript',
      should: 'generate correct files',
      snap: getSortedFilePaths(root)
    })

    await assert({
      given: 'esops_react_native_typescript',
      should: 'generate correct package.json',
      snap: getFileContents(path.join(root, 'package.json'))
    })

    await assert({
      given: 'esops_react_native_typescript',
      should: 'generate correct .gitignore',
      snap: getFileContents(path.join(root, '.gitignore'))
    })
  })
})
