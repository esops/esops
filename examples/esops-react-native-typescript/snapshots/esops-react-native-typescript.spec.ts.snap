module.exports[`Describe esops react native typescript | Given esops_react_native_typescript: should generate correct files`] = `

[
  ".buckconfig",
  ".flowconfig",
  ".gitattributes",
  ".gitignore",
  ".watchmanconfig",
  "App.js",
  "__tests__",
  "__tests__/App-test.js",
  "android",
  "android/app",
  "android/app/BUCK",
  "android/app/build.gradle",
  "android/app/build_defs.bzl",
  "android/app/proguard-rules.pro",
  "android/app/src",
  "android/app/src/debug",
  "android/app/src/debug/AndroidManifest.xml",
  "android/app/src/main",
  "android/app/src/main/AndroidManifest.xml",
  "android/app/src/main/java",
  "android/app/src/main/java/com",
  "android/app/src/main/java/com/testproject",
  "android/app/src/main/java/com/testproject/MainActivity.java",
  "android/app/src/main/java/com/testproject/MainApplication.java",
  "android/app/src/main/res",
  "android/app/src/main/res/mipmap-hdpi",
  "android/app/src/main/res/mipmap-hdpi/ic_launcher.png",
  "android/app/src/main/res/mipmap-hdpi/ic_launcher_round.png",
  "android/app/src/main/res/mipmap-mdpi",
  "android/app/src/main/res/mipmap-mdpi/ic_launcher.png",
  "android/app/src/main/res/mipmap-mdpi/ic_launcher_round.png",
  "android/app/src/main/res/mipmap-xhdpi",
  "android/app/src/main/res/mipmap-xhdpi/ic_launcher.png",
  "android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.png",
  "android/app/src/main/res/mipmap-xxhdpi",
  "android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png",
  "android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.png",
  "android/app/src/main/res/mipmap-xxxhdpi",
  "android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png",
  "android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png",
  "android/app/src/main/res/values",
  "android/app/src/main/res/values/strings.xml",
  "android/app/src/main/res/values/styles.xml",
  "android/build.gradle",
  "android/gradle",
  "android/gradle.properties",
  "android/gradle/wrapper",
  "android/gradle/wrapper/gradle-wrapper.jar",
  "android/gradle/wrapper/gradle-wrapper.properties",
  "android/gradlew",
  "android/gradlew.bat",
  "android/keystores",
  "android/keystores/BUCK",
  "android/keystores/debug.keystore.properties",
  "android/settings.gradle",
  "app.json",
  "babel.config.js",
  "index.js",
  "ios",
  "ios/TestProject",
  "ios/TestProject-tvOS",
  "ios/TestProject-tvOS/Info.plist",
  "ios/TestProject-tvOSTests",
  "ios/TestProject-tvOSTests/Info.plist",
  "ios/TestProject.xcodeproj",
  "ios/TestProject.xcodeproj/project.pbxproj",
  "ios/TestProject.xcodeproj/xcshareddata",
  "ios/TestProject.xcodeproj/xcshareddata/xcschemes",
  "ios/TestProject.xcodeproj/xcshareddata/xcschemes/TestProject-tvOS.xcscheme",
  "ios/TestProject.xcodeproj/xcshareddata/xcschemes/TestProject.xcscheme",
  "ios/TestProject/AppDelegate.h",
  "ios/TestProject/AppDelegate.m",
  "ios/TestProject/Base.lproj",
  "ios/TestProject/Base.lproj/LaunchScreen.xib",
  "ios/TestProject/Images.xcassets",
  "ios/TestProject/Images.xcassets/AppIcon.appiconset",
  "ios/TestProject/Images.xcassets/AppIcon.appiconset/Contents.json",
  "ios/TestProject/Images.xcassets/Contents.json",
  "ios/TestProject/Info.plist",
  "ios/TestProject/main.m",
  "ios/TestProjectTests",
  "ios/TestProjectTests/Info.plist",
  "ios/TestProjectTests/TestProjectTests.m",
  "metro.config.js",
  "package.json",
  "yarn.lock"
]`

module.exports[`Describe esops react native typescript | Given esops_react_native_typescript: should generate correct package.json`] = `

{
  "name": "TestProject",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "test": "jest"
  },
  "dependencies": {
    "react": "16.8.3",
    "react-native": "0.59.8"
  },
  "devDependencies": {
    "@babel/core": "^7.4.4",
    "@babel/runtime": "^7.4.4",
    "babel-jest": "^24.8.0",
    "jest": "^24.8.0",
    "metro-react-native-babel-preset": "^0.54.1",
    "react-test-renderer": "16.8.3"
  },
  "jest": {
    "preset": "react-native"
  }
}
`

module.exports[`Describe esops react native typescript | Given esops_react_native_typescript: should generate correct .gitignore`] = `

# OSX
#
.DS_Store

# Xcode
#
build/
*.pbxuser
!default.pbxuser
*.mode1v3
!default.mode1v3
*.mode2v3
!default.mode2v3
*.perspectivev3
!default.perspectivev3
xcuserdata
*.xccheckout
*.moved-aside
DerivedData
*.hmap
*.ipa
*.xcuserstate
project.xcworkspace

# Android/IntelliJ
#
build/
.idea
.gradle
local.properties
*.iml

# node.js
#
node_modules/
npm-debug.log
yarn-error.log

# BUCK
buck-out/
\.buckd/
*.keystore

# fastlane
#
# It is recommended to not store the screenshots in the git repo. Instead, use fastlane to re-generate the
# screenshots whenever they are needed.
# For more information about the recommended setup visit:
# https://docs.fastlane.tools/best-practices/source-control/

*/fastlane/report.xml
*/fastlane/Preview.html
*/fastlane/screenshots

# Bundle artifact
*.jsbundle

### ESOPS AUTO GENERATED BEGIN ###
/.buckconfig
/.flowconfig
/.gitattributes
/.watchmanconfig
/App.js
/__tests__/App-test.js
/android/app/BUCK
/android/app/build.gradle
/android/app/build_defs.bzl
/android/app/proguard-rules.pro
/android/app/src/debug/AndroidManifest.xml
/android/app/src/main/AndroidManifest.xml
/android/app/src/main/java/com/testproject/MainActivity.java
/android/app/src/main/java/com/testproject/MainApplication.java
/android/app/src/main/res/mipmap-hdpi/ic_launcher.png
/android/app/src/main/res/mipmap-hdpi/ic_launcher_round.png
/android/app/src/main/res/mipmap-mdpi/ic_launcher.png
/android/app/src/main/res/mipmap-mdpi/ic_launcher_round.png
/android/app/src/main/res/mipmap-xhdpi/ic_launcher.png
/android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.png
/android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png
/android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.png
/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png
/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png
/android/app/src/main/res/values/strings.xml
/android/app/src/main/res/values/styles.xml
/android/build.gradle
/android/gradle.properties
/android/gradle/wrapper/gradle-wrapper.jar
/android/gradle/wrapper/gradle-wrapper.properties
/android/gradlew
/android/gradlew.bat
/android/keystores/BUCK
/android/keystores/debug.keystore.properties
/android/settings.gradle
/app.json
/babel.config.js
/index.js
/ios/TestProject-tvOS/Info.plist
/ios/TestProject-tvOSTests/Info.plist
/ios/TestProject.xcodeproj/project.pbxproj
/ios/TestProject.xcodeproj/xcshareddata/xcschemes/TestProject-tvOS.xcscheme
/ios/TestProject.xcodeproj/xcshareddata/xcschemes/TestProject.xcscheme
/ios/TestProject/AppDelegate.h
/ios/TestProject/AppDelegate.m
/ios/TestProject/Base.lproj/LaunchScreen.xib
/ios/TestProject/Images.xcassets/AppIcon.appiconset/Contents.json
/ios/TestProject/Images.xcassets/Contents.json
/ios/TestProject/Info.plist
/ios/TestProject/main.m
/ios/TestProjectTests/Info.plist
/ios/TestProjectTests/TestProjectTests.m
/metro.config.js
/package.json
/yarn.lock
### ESOPS AUTO GENERATED END ###
`

